### PR TITLE
cal 485 prevent users from changing their username to premium ones

### PR DESCRIPTION
**Pull request by zomars**
This is my first attempt for using tRCP. @KATT any feedback is appreciated. 🙏🏽

### CODE REVIEW

**Comment by PeerRich** [comment]
lgtm 

image

**Review by PeerRich — APPROVED** [review]

**Review by pumfleet — APPROVED** [review]
This looks good to me, just want to get @KATT's approval before merging, as we're all new to tRPC

File: server/routers/viewer.tsx
Comment (KATT) [inline server/routers/viewer.tsx:7]: These two should probably be named distinctly different things to avoid confusion

File: server/routers/viewer.tsx
Comment (KATT) [inline server/routers/viewer.tsx:104]: We don't need to construct if they follow the same name
```suggestion
      const data: Prisma.UserUpdateInput = {
        ...input,
      };
```

File: server/routers/viewer.tsx
Comment (KATT) [inline server/routers/viewer.tsx:81]: align with db
```suggestion
      bio: z.string().optional(),
```

**Review by KATT — CHANGES_REQUESTED** [review]
Some more comments apart from #801

File: server/routers/viewer.tsx
Comment (KATT) [reply server/routers/viewer.tsx:7]: As in `"@ee/lib/core/checkUsername"` // `"@lib/core/checkUsername"`

Also, I'm not a big fan of `export default` -- I prefer named exports

Comment (zomars) [reply server/routers/viewer.tsx:7]: Agreed

**Comment by zomars** [comment]
Also added an UX improvement when you try to update, it auto scrolls to the error warning: https://www.loom.com/share/b25e2d128d84445ab008abea0e594456

**Review by KATT — APPROVED** [review]

**Comment by KATT** [comment]
Make sure to test it on a preview link before deploying as you have some deployment-specific logic

**Comment by PeerRich** [comment]
> Make sure to test it on a preview link before deploying as you have some deployment-specific logic

@zomars all good? can we merge?

**Comment by zomars** [comment]
Testes on prod and it's working 👍🏽

**Pull request merged**
